gobrew.go

```diff
@@ -671,42 +676,64 @@ func (gb *GoBrew) getGithubTags(repo string) (result []string) {
 
 	githubTags = make(map[string][]string, 0)
 	client := &http.Client{}
-	url := "https://api.github.com/repos/kevincobain2000/gobrew/git/refs/tags"
-	if repo == "golang/go" {
+	url := goTagsApi
+	if repo == golangRepo {
 		url = goBrewTagsApi
 	}
+
 	request, err := http.NewRequest("GET", url, nil)
 	if err != nil {
 		utils.Errorf("[Error] Cannot create request: %s", err)
 		return
 	}
 
-	request.Header.Set("User-Agent", "gobrew")
+	request.Header.Set("User-Agent", gobrewRepo)
```

Comment:
We can set Header token:
https://docs.github.com/en/rest/overview/authenticating-to-the-rest-api?apiVersion=2022-11-28

> Note: In most cases, you can use Authorization: Bearer or Authorization: token to pass a token. 

Like this:
```golang
	if token := os.Getenv("GITHUB_TOKEN"); token != "" {
		request.Header.Set("Authorization", "Bearer "+token)
	}
```

For github actions this will allow you to expand the limits.

I think we could create new function for request from github. Or just use https://github.com/carlmjohnson/requests

Reply:
Ratelimits for personal accounts:
https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limits-for-requests-from-personal-accounts

> User access token requests are limited to 5,000 requests per hour and per authenticated user. All requests from OAuth applications authorized by a user or a personal access token owned by the user, and requests authenticated with any of the user's authentication credentials, share the same quota of 5,000 requests per hour for that user.

> For unauthenticated requests, the rate limit allows for up to 60 requests per hour. Unauthenticated requests are associated with the originating IP address, and not the person making requests.

If we will use token on requests, we can increase limits to 83 times.

Github Actions use `GITHUB_TOKEN` variable for storing personal token for each run. And we can use it in requests.

I think the token will not be needed for normal use. But we can provide information about it in our README.

Reply:
I mean normal user will never see this issue. Nor in the actions as well. 

The rate limit only effects the gobrew self-update. Other go installs are rate limit proof. 

This caching is just a measure. Although now I am a bit reluctant to merge this, as it doesn’t bring any value.

Reply:
OK, got it :)